python/ray/tests/test_failure_3.py

```diff
@@ -347,6 +352,103 @@ def pid(self):
         ray.get(t)
 
 
+def test_no_worker_child_process_leaks(ray_start_cluster, tmp_path):
```

Comment:
is it possible to add a test with torch data loader?

Reply:
Sure, let me try today

Reply:
Still can't reproduce (torch shuts down successfully whenever I leak the processes). https://github.com/ray-project/ray/issues/31451#issuecomment-1463185117

Already spent a lot of time on this a few weeks ago, let's see if this fixes the issue in the user environment.